File: mobile/android/app/src/main/java/com/enkodu/companion/data/TransferState.kt
package com.enkodu.companion.data

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "transfers",
    indices = [
        Index(value = ["uploadId"], unique = true),
        Index(value = ["jobId"], unique = true),
        Index(value = ["status"])
    ]
)
data class TransferState(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val uploadId: String? = null,
    val jobId: String? = null,
    val filePath: String,
    val localTempPath: String? = null,
    val totalBytes: Long = 0,
    val bytesTransferred: Long = 0,
    val status: String = TransferStatus.PENDING.name,
    val transferType: String = TransferType.UPLOAD.name,
    val lastError: String? = null,
    val retryCount: Int = 0,
    val networkType: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

enum class TransferStatus {
    PENDING, ACTIVE, PAUSED, FAILED, DONE, CANCELLED
}

enum class TransferType {
    UPLOAD, DOWNLOAD
}

File: mobile/ios/Enkodu/Enkodu/Data/TransferState.swift
import Foundation
import CoreData

@objc(TransferState)
public class TransferState: NSManagedObject {
    @NSManaged public var id: UUID
    @NSManaged public var uploadId: String?
    @NSManaged public var jobId: String?
    @NSManaged public var filePath: String
    @NSManaged public var localTempPath: String?
    @NSManaged public var totalBytes: Int64
    @NSManaged public var bytesTransferred: Int64
    @NSManaged public var status: String
    @NSManaged public var transferType: String
    @NSManaged public var lastError: String?
    @NSManaged public var retryCount: Int16
    @NSManaged public var createdAt: Date
    @NSManaged public var updatedAt: Date
}

extension TransferState {
    @nonobjc public class func fetchRequest() -> NSFetchRequest<TransferState> {
        return NSFetchRequest<TransferState>(entityName: "TransferState")
    }
}

enum TransferStatus: String {
    case pending = "PENDING"
    case active = "ACTIVE"
    case paused = "PAUSED"
    case failed = "FAILED"
    case done = "DONE"
    case cancelled = "CANCELLED"
}

enum TransferType: String {
    case upload = "UPLOAD"
    case download = "DOWNLOAD"
}
